include/core/LayerData.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef __NGEN_LAYER_DATA__
+#define __NGEN_LAYER_DATA__
+
+#include <unordered_map>
+#include <string>
+#include <vector>
+
+namespace ngen
+{
+    /**
+     * @brief A simple structure to hold meta data realated to a computational layer
+    */
+    
+    struct LayerDescription
+    {
+        std::string name;             //> The name of the computational layer
+        std::string time_step_units;  //> The time units that time_step is expressed in
+        int id;                       //> The layer id which determins the order that this layer will be processed in
+        double time_step;             //> The time step that will be used for models in this layer
+    };
+    
+    /**
+     * @brief A class to hold a manage layer metadata for the system
+    */
+
+    class LayerDataStorage
+    {
+        public:
+
+        LayerDataStorage() {}
+
+        ~LayerDataStorage() {}
+
+        const LayerDescription& get_layer(int id) const 
+        { 
+            return stored_layers.at(id); 
+        }
+
+        void put_layer(const LayerDescription& desc, int id) 
+        {
+             stored_layers[id] = desc;
+             keys.push_back(id);
+        }
+
+        bool exists(int id) 
+        { 
+            return stored_layers.find(id) != stored_layers.end(); 
+        }
+
+        std::vector<int>& get_keys() { return keys; }
+
+        private:
+
+        std::unordered_map<int,LayerDescription> stored_layers;
+        std::vector<int> keys;
```

Comment:
The map will be accessed a lot. Unorderedmap is O(1) vs O(lg(N)) of map, particularly when there will be a small number of keys. The use of vector is also not ideal because not all layer slots will be filled.

Currently layers range in id from -20 (lowest ground water) to 80 (upper atmosphere) but few ids will be used at any time. This means allowing retrieving a layer with its id requires a map (or an index array), since the list of keys is both small and will be needed by the main execution loop there is no reason not to build the key list while layers are added, and then store it for reference.